fishjam/api/_fishjam_client.py

```diff
@@ -0,0 +1,167 @@
+"""
+Fishjam client used to manage rooms
+"""
+
+from dataclasses import dataclass
+from typing import List, Literal, NewType, Tuple
+
+from fishjam._openapi_client.api.room import add_peer as room_add_peer
+from fishjam._openapi_client.api.room import create_room as room_create_room
+from fishjam._openapi_client.api.room import delete_peer as room_delete_peer
+from fishjam._openapi_client.api.room import delete_room as room_delete_room
+from fishjam._openapi_client.api.room import get_all_rooms as room_get_all_rooms
+from fishjam._openapi_client.api.room import get_room as room_get_room
+from fishjam._openapi_client.models import (
+    AddPeerJsonBody,
+    Peer,
+    PeerOptionsWebRTC,
+    RoomConfig,
+)
+from fishjam._openapi_client.models.peer_options_web_rtc_metadata import (
+    PeerOptionsWebRTCMetadata,
+)
+from fishjam._openapi_client.models.room_config_video_codec import RoomConfigVideoCodec
+from fishjam.api._client import Client
+
+PeerToken = NewType("PeerToken", str)
+
+
+@dataclass
+class Room:
+    """Description of the room state"""
+
+    config: RoomConfig
+    """Room configuration"""
+    id: str
+    """Room ID"""
+    peers: List[Peer]
+    """List of all peers"""
+
+
+@dataclass
+class RoomOptions:
+    """Description of a room options"""
+
+    max_peers: int = None
+    """Maximum amount of peers allowed into the room"""
+    peer_disconnected_timeout: int = None
+    """
+    Duration (in seconds) after which the peer will be removed if it is disconnected.
+    If not provided, this feature is disabled.
+    """
+    peerless_purge_timeout: int = None
+    """
+    Duration (in seconds) after which the room will be removed 
+    if no peers are connected. If not provided, this feature is disabled.
+    """
+    room_id: str = None
+    """
+    Custom id used for identifying room within Fishjam.
+    Must be unique across all rooms. If not provided, random UUID is generated.
+    """
+    video_codec: Literal["h264", "vp8"] = None
+    """Enforces video codec for each peer in the room"""
+    webhook_url: str = None
+    """URL where Fishjam notifications will be sent"""
+
+
+@dataclass
+class PeerOptions:
+    """Options specific to the Peer"""
+
+    enable_simulcast: bool = True
+    """Enables the peer to use simulcast"""
+    metadata: dict = None
+    """Peer metadata"""
+
+
+class FishjamClient(Client):
+    """Allows for managing rooms"""
+
+    def __init__(self, fishjam_url: str, management_token: str):
+        """
+        Create FishjamClient instance, providing the fishjam url and managment token.
+        """
+        super().__init__(fishjam_url=fishjam_url, management_token=management_token)
+
+    def create_peer(
+        self, room_id: str, options: PeerOptions = PeerOptions()
```

Comment:
**issue:** `options` should have default value  `None` and should be set to `PeerOptions` inside of the function, because `PeerOptions.metadata` is mutable